src/invariant/InvariantManagerImpl.cpp

```diff
@@ -148,10 +170,13 @@ InvariantManagerImpl::enableInvariant(std::string const& name)
 void
 InvariantManagerImpl::onInvariantFailure(
         std::shared_ptr<Invariant> invariant,
-        std::string const& message) const
+        std::string const& message,
+        uint32_t ledger) const
 {
-    mMetricsRegistry.NewCounter({"invariant", "does-not-hold",
+    mMetricsRegistry.NewCounter({"invariant", "does-not-hold", "count",
                                  invariant->getName()}).inc();
+    mMetricsRegistry.NewCounter({"invariant", "does-not-hold", "on-ledger",
```

Comment:
I'd prefer to not use metrics as the mechanism to capture a snapshot for "info" (the `on-ledger` metric is just bloating metrics):
The way we propagate status in the history subsystem for example, it by simply saving the last failure in a member variable. Here all you'd need to do is store the `<message, ledger>` pair in a `mLastFailure` member. Doing like this will not tie us up to the limitations of the metrics subsystem while at the same time keeping the door open to storing more than the very last failure.